ui/src/main/java/org/pentaho/di/ui/trans/steps/fileinput/text/TextFileCSVImportProgressDialog.java

```diff
@@ -275,8 +276,15 @@ private String doScan( IProgressMonitor monitor, final boolean failOnParseError
     if ( meta.content.header ) {
       int skipped = 0;
       while ( line != null && skipped < meta.content.nrHeaderLines ) {
-        line = TextFileInputUtils.getLine( log, reader, encodingType, fileFormatType, lineBuffer );
-        skipped++;
+        if ( TextFileInputUtils.checkPattern( line, meta.getEnclosure() ) % 2 == 0 ) {
```

Comment:
Check that the number of enclosures in a line is even. If not even it means that there was an enclosed line break. We need to read the next like(s) to get the remaining data in this row.